Capstone Project/ViewControllers/Workouts/CurrentWorkoutViewController.m

```diff
@@ -84,14 +84,65 @@ -(void)getExercises{
 }
 
 - (IBAction)didTapAdd:(id)sender {
-    NSArray *repsPerSet = [self.repsTextField.text componentsSeparatedByString:@"/"];
-    [Exercise newExercise:self.searchBar.text withWeight:[self.weightTextField.text floatValue] withSet1:[[repsPerSet objectAtIndex:0] integerValue] withSet2:[[repsPerSet objectAtIndex:1] integerValue] withSet3:[[repsPerSet objectAtIndex:2] integerValue] withSet4:[[repsPerSet objectAtIndex:3] integerValue] withSet5:[[repsPerSet objectAtIndex:4] integerValue] withWorkout:self.selectedWorkout withCompletion:^(BOOL succeeded, NSError * _Nullable error){
-        if(!error){
-            [self getExercises];
+    //Check for any blank fields
+    if([self.searchBar.text isEqual:@""] || [self.weightTextField.text isEqual:@""] || [self.repsTextField.text isEqual:@""]){
+        self.errorLabel.text = @"One or more fields are Blank";
+    }
+    else if ([self.weightTextField.text floatValue] <= 0){
+        self.errorLabel.text = @"The weight is not valid";
+    }
+    else{
+        //Getting the array of the sets
+        NSArray *repsPerSet = [self.repsTextField.text componentsSeparatedByString:@"/"];
+        //Checking if sets are valid
+        if(![self areSetsValid:repsPerSet]){
+            self.errorLabel.text = @"Input not valid";
+        }else{
+            //Need to make the array mutable to add ceros
+            NSMutableArray *repsPerSetMut = [(NSArray *)repsPerSet mutableCopy];
+            //Adding 0 if set is empty
+            while(repsPerSetMut.count < 5){
+                [repsPerSetMut addObject: @"0"];
+            }
+            [Exercise newExercise:self.searchBar.text withWeight:[self.weightTextField.text floatValue] withSet1:[[repsPerSetMut objectAtIndex:0] intValue] withSet2:[[repsPerSetMut objectAtIndex:1] intValue] withSet3:[[repsPerSetMut objectAtIndex:2] intValue] withSet4:[[repsPerSetMut objectAtIndex:3] intValue] withSet5:[[repsPerSetMut objectAtIndex:4] intValue] withWorkout:self.selectedWorkout withCompletion:^(BOOL succeeded, NSError * _Nullable error){
```

Comment:
What if the size of the array is less than 5. The number of sets done on that day is less. If that happens objectAtIndex:4 call will results in crash. Using it without knowing the actual size of the array is a common cause for crash